### PR TITLE
Detect network changes and dispatch

### LINKED ISSUE #428: Detect network changes
When using a web3 provider (metamask or...) it is important to be able to detect network changes (from main net to rinkeby to... etc) and then delete the localStorage state because it cannot be applied to that network.

### PR DESCRIPTION
**Pull request by cellog**
# Description

This fixes #428 by creating a middleware that polls to detect network changes every second. If detected, it dispatches a redux action to update the store

An unresolved question: This action should trigger the lock middleware to reset the account.
Currently that only happens on a `network.changed` event. Should I move that code so that
both locations will trigger the account reset as well?

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [X] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    const middleware = require('../../middlewares/networkVersionDetectMiddleware')
+      .default
+    const store = {
+      dispatch: jest.fn(),
+      getState() {
+        return { network: { name: 'nothi' } }
+      },
+    }
+
+    window.web3 = {
```
[the remark this answers is not in the document]
Comment (cellog) [reply unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js]: I tried putting these into beforeEach and got some *really* weird error messages, so I'm going to leave these untouched. Other things are fixed

[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js]: Hum... I am worried about your comment. Can you tell me more about the issues?

[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js]: Hum... I am worried about your comment. Can you tell me more about the issues?

[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js]: Hum... I am worried about your comment. Can you tell me more about the issues?

[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js]: Why is that 3 comments???

I think the challenges are probably due to the use of `window`. Can you replace that with `environment` and see what happens?

[the remark this answers is not in the document]
Comment (cellog) [reply unlock-app/src/__tests__/middlewares/networkVersionDetectMiddleware.test.js]: good news! The error was a configuration issue locally. Pushing a working dry-er version now.

**Pull request merged**